components/job-orchestration/job_orchestration/executor/compress/fs_compression_task.py

```diff
@@ -76,6 +79,17 @@ def make_clp_s_command(
     return compression_cmd
 
 
+def update_tags(db_conn, db_cursor, table_prefix, archive_id, tags):
+    for tag in tags:
+        db_cursor.execute(f"INSERT IGNORE INTO {table_prefix}tags (tag_name) VALUES (%s)", (tag,))
```

Comment:
This seems like it would be expensive when we're creating a lot of archives in parallel. Would it make more sense to create the tags ahead of time in the scheduler (controller) and then pass the tag IDs down to the workers? It's possible we don't create any archives corresponding to the tags we insert, but that seems like it would be rare and a small cost to pay.